src/Features/model/UniversalSkillsLogic/UniversalSkillsLogic.tsx

```diff
@@ -0,0 +1,181 @@
+import {toast} from 'react-toastify'
+import {useEffect, useState} from 'react'
+import useSkillsData from './hooks/useSkillsData'
+import {TransformedArray} from '@/Features'
+import useFormData from './hooks/useFormData'
+import {MasteryOptions} from './utilits/MasteryOptions'
+import useSkillsCvData from './hooks/useSkillsCvData'
+import {useBreadCrumbsContext} from '@/App'
+
+interface techno {
+  category: string
+  categoryId: string
+  mastery: string
+  name: string
+  proficiency: number
+}
+
+export interface filterData {
+  category: string
+  technologies: [techno]
+}
+
+export type FiltersTechnologies = filterData
+
+interface UniversalSkillsPageProps {
+  forState: 'cv' | 'skills'
+  userId: string
+  handleDeleteSkill: (arr: Set<string>) => Promise<void>
+  handleAddSkill: (
+    skill: string,
+    skillMaster: string,
+    categoryId: string
+  ) => Promise<void>
+  handleUpdateSkill: (
+    skill: string,
+    skillMaster: string,
+    categoryId: string
+  ) => Promise<void>
+}
+
+const UniversalSkillsLogic = ({
+  forState,
+  userId,
+  handleDeleteSkill,
+  handleAddSkill,
+  handleUpdateSkill,
+}: UniversalSkillsPageProps) => {
+  const {t} = useTranslation()
+  const [openAdd, setOpen] = useState(false)
+  const breadcrumb = useBreadCrumbsContext()
+
+  const useSkillsHook = forState === 'skills' ? useSkillsData : useSkillsCvData
+  const {
+    error,
+    groupedData,
+    loading,
+    transformedSkills,
+    userSkillsData,
+    refetch,
+    name,
+  } = useSkillsHook(userId)
+
+  const transformArray: TransformedArray[] = []
+  transformedSkills.map((elem) => {
+    transformArray.push({value: 'notSelect', label: elem.category})
+    transformArray.push(...elem.technologies)
+  })
+
+  const {formData, handleOpenAdd, handleOpenEdit} = useFormData(transformArray)
+
+  const handleClose = () => {
+    setOpen(false)
+  }
+
+  const handleAdd = async (skill: string, skillMaster: string) => {
+    const techno = formData.firstSelectOptions.filter(
+      (elem) => elem.label === skill
+    )
+    if (!techno[0].id) {
+      techno[0].id = ''
+    }
+    try {
+      await handleAddSkill(skill, skillMaster, techno[0].id)
+      handleClose()
+      await refetch()
+    } catch (err) {
+      console.error(err)
+    }
+  }
+
+  const handleDelete = async (arr: Set<string>) => {
+    try {
+      await handleDeleteSkill(arr)
+      await refetch()
+    } catch (err) {
+      console.error(err)
+    }
+  }
+
+  const handleUpdate = async (skill: string, skillMaster: string) => {
+    const techno = formData.firstSelectOptions.filter(
+      (elem) => elem.label === skill
+    )
+    if (!techno[0].id) {
+      techno[0].id = ''
+    }
+    try {
+      await handleUpdateSkill(skill, skillMaster, techno[0].id)
+      handleClose()
+      await refetch()
+    } catch (err) {
+      console.error(err)
+    }
+  }
+
+  useEffect(() => {
+    if (name) {
+      breadcrumb.setCurrentBread(name)
+    }
+  })
```

Comment:
`useEffect` без зависимостей
```suggestion
  useEffect(() => {
    if (name) {
      breadcrumb.setCurrentBread(name)
    }
  }, [name])
```